testtools/matchers/_impl.py

```diff
@@ -53,7 +62,49 @@ def __str__(self):
         raise NotImplementedError(self.__str__)
 
 
-class Mismatch(object):
+Matcher = IMatcher
+
+
+class IMismatch(object):
+    """A mismatch detected by an IMatcher.
+
+    Describes the protocol that mismatches are required to implement.
+    """
+
+    # XXX: jml would like to extend this interface to include the thing that
```

Comment:
>  does "include the thing that" mean "the matchee argument passed to match()"

Yes.

> Why do we need to pass it to describe()?

Did I propose this?

> An example of the "composition" use case you allude to would be appreciated.

Sorry, the intervening 13 months have somehow scrubbed it from my memory. However, looking at the [file linked to in my comment](https://github.com/ClusterHQ/flocker/blob/master/flocker/testtools/matchers.py#L176) provides just such an example. You'll have to figure out what it does, I'm afraid.